front/src/app/feature/admin-management/components/team/navbar-team-page/navbar-team-page.component.ts

```diff
@@ -0,0 +1,167 @@
+import { Component, Input, OnInit, OnChanges, SimpleChanges, OnDestroy } from '@angular/core';
+import { NavigationEnd, Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+import {TeamMemberService} from '../../../services/team/team-member.service';
+import {AuthService} from '../../../../../core/login/services/auth.service';
+import {UserRoleService} from '../../../services/team/user-role.service';
+import {NavbarAdminPageComponent} from '../../navbar-admin-page/navbar-admin-page.component';
+import {NavbarConfig} from '../../../type/components/navbar-config';
+
+@Component({
+  selector: 'app-navbar-team-page',
+  standalone: true,
+  imports: [
+    NavbarAdminPageComponent
+  ],
+  templateUrl: './navbar-team-page.component.html',
+  styleUrl: './navbar-team-page.component.scss'
+})
+export class NavbarTeamPageComponent implements OnInit, OnChanges, OnDestroy {
+  @Input() teamUrl: string = '';
+  @Input() teamId: string = '';
+  @Input() teamName: string = '';
+  @Input() userRole: string = '';
+
+  activePage: string = '';
+  currentUserRole: string = 'Member';
+  navbarConfig: NavbarConfig = { leftButtons: [] };
+
+  private userSubscription?: Subscription;
+  private roleSubscription?: Subscription;
+  private routerSubscription?: Subscription;
+  private currentUser: any = null;
+
+  constructor(
+    private router: Router,
+    private teamMemberService: TeamMemberService,
+    private authService: AuthService,
+    private userRoleService: UserRoleService
+  ) {}
+
+  ngOnInit(): void {
+    this.setupNavbarConfig();
+    this.setActivePage();
+
+    this.routerSubscription = this.router.events.subscribe(event => {
+      if (event instanceof NavigationEnd) {
+        this.setActivePage();
+      }
+    });
+
+    this.userSubscription = this.authService.user$.subscribe(user => {
+      this.currentUser = user;
+      if (user && this.teamId) {
+        this.loadUserRole(user.uid);
+      }
+    });
+
+    this.userRoleService.getRole().subscribe(role => {
+      if (role) {
+        this.currentUserRole = role;
+      }
+    });
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['userRole']?.currentValue) {
+      this.currentUserRole = changes['userRole'].currentValue;
```

Comment:
todo: cette ligne ne sert à rien, une passé en Signal ça donne ça 

```suggestion
      currentUserRole = computed(() => this.userRole());
```